lib/diff.ml

```diff
@@ -123,39 +143,169 @@ let rec type_item ~typing_env ~name ~reference ~current =
       in
       match (type_coercion1 (), type_coercion2 ()) with
       | None, None -> None
-      | _, _ -> (
+      | ref_mismatch, cur_mismatch -> (
           match (reference.type_kind, current.type_kind) with
-          | Type_record (ref_label_lst, _), Type_record (cur_label_lst, _) ->
-              let changed_lbls =
-                modified_record_type ~typing_env ~ref_label_lst ~cur_label_lst
-              in
-              Some
-                (Type { tname = name; tdiff = Modified (Compound changed_lbls) })
-          | _, _ ->
+          | Type_record (ref_label_lst, _), Type_record (cur_label_lst, _) -> (
+              match (ref_mismatch, cur_mismatch) with
+              | ( Some (Includecore.Record_mismatch _),
```

Comment:
I'm not sure that's the right approach here, what if the types differ on multiple levels and Includecore returns one of the other reasons? We'd still want to show that diff.

We have to trust that our function does the right thing. Compute the diff between the two records, if it's `[]` then return `None`, otherwise returns the list of diffs!

Reply:
This applies for the record case below as well.